### PR TITLE
Multiple property filter improvements (weapon damage, armour filters, etc.)

### PR DESCRIPTION
**Pull request by warxhead1**
# Trade UI and Filtering Improvements

This PR updates the trade search interface and filtering functionality to more closely match the official Path of Exile trade site.

### UI Improvements
- Reorganized filter layout to match official trade site
- Added proper separator placement between filter sections
- Removed redundant headers for cleaner interface
- Fixed corrupted status display in trade results

### Filtering Enhancements
- Added type and rarity filter controls
  - Type filter enabled by default
  - Option to search by specific type or base category
- Fixed category handling to use proper API query paths
- Improved base category display 

### Code Quality
- Maintained consistent code style
- Added proper error handling
- Ensured type safety throughout changes
- Removed redundant code

## Testing
- Verified type filtering works with both specific and base types
- Confirmed rarity filtering matches official site behavior
- Tested corrupted item display in search results
- Validated separator placement and UI layout
- Confirmed API queries match official site format

## Screenshots
old_sidekick_interface
new_sidekick_interface

image

### CODE REVIEW

**Comment by leMicin** [comment]
In addition to the above:
- Added colours to the trade result properties
- Armour filters work

**Review by domialex — COMMENTED** [review]
`trade_api_filters.json` and `trade_api_response.json` are leftovers?

File: src/Sidekick.Apis.Poe/Parser/ItemParser.cs [hunk 1]
```diff
@@ -123,7 +123,12 @@ private Header ParseHeader(ParsingItem parsingItem)
 
             if (firstLine.StartsWith(gameLanguageProvider.Language.Classes.Prefix))
             {
-                var classLine = firstLine.Replace(gameLanguageProvider.Language.Classes.Prefix, "").Trim();
+                var classLine = firstLine.Replace(gameLanguageProvider.Language.Classes.Prefix, "").Trim(' ', ':');
+                if (classLine == "Misc Map Items")
```
[the remark this answers is not in the document]
Comment (leMicin) [reply src/Sidekick.Apis.Poe/Parser/ItemParser.cs:127]: No, english has a weird thing where the API says Map Fragment and the game says Misc Map Items. I thought we could hardcode it here

File: src/Sidekick.Common/Game/Languages/Implementations/GameLanguageDe.cs [hunk 2]
```diff
@@ -49,6 +49,14 @@ public class GameLanguageDe : IGameLanguage
 
     public string DescriptionElementalDamage => "Elementarschaden";
 
+    public string DescriptionFireDamage => "__";
```
[the remark this answers is not in the document]
Comment (leMicin) [reply src/Sidekick.Common/Game/Languages/Implementations/GameLanguageDe.cs:52]: Will translate these

File: src/Sidekick.Common/Game/Languages/Implementations/GameLanguageZhTw.cs [hunk 3]
```diff
@@ -47,6 +47,14 @@ public class GameLanguageZhTw : IGameLanguage
 
     public string DescriptionElementalDamage => "元素傷害";
 
+    public string DescriptionFireDamage => "__";
```
[the remark this answers is not in the document]
Comment (leMicin) [reply src/Sidekick.Common/Game/Languages/Implementations/GameLanguageZhTw.cs:50]: I will translate

**Pull request merged**